test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1424,4 +1424,243 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
+      //     keyVaultNamespace: "keyvault.datakeys";
+      //     kmsProviders: { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } }
+      //   }
+      // Configure ``encryptedClient`` with "retryReads=false".
+      encryptedClient = this.configuration.newClient(
+        {},
+        {
+          retryReads: false,
+          monitorCommands: true,
+          autoEncryption: {
+            keyVaultNamespace: 'keyvault.datakeys',
+            kmsProviders: getKmsProviders(LOCAL_KEY)
+          }
+        }
+      );
+      // Register a listener for CommandSucceeded events on ``encryptedClient``.
+      encryptedClient.on('commandSucceeded', event => {
+        if (event.commandName === 'aggregate') {
+          aggregateSucceeded = event;
+        }
+      });
+      // The listener must store the most recent CommandStartedEvent reply for the "aggregate" command.
+      encryptedClient.on('commandStarted', event => {
+        if (event.commandName === 'aggregate') {
+          aggregateStarted = event;
+        }
+      });
+      // The listener must store the most recent CommandFailedEvent error for the "aggregate" command.
+      encryptedClient.on('commandFailed', event => {
+        if (event.commandName === 'aggregate') {
+          aggregateFailed = event;
+        }
+      });
+    });
+
+    afterEach(async function () {
+      aggregateSucceeded = undefined;
+      aggregateStarted = undefined;
+      aggregateFailed = undefined;
+      await setupClient.close();
+      await encryptedClient.close();
+    });
+
+    context('Case 1: Command Error', function () {
+      beforeEach(async function () {
+        // Use ``setupClient`` to configure the following failpoint:
+        //    {
+        //         "configureFailPoint": "failCommand",
+        //         "mode": {
+        //             "times": 1
+        //         },
+        //         "data": {
+        //             "errorCode": 123,
+        //             "failCommands": [
+        //                 "aggregate"
+        //             ]
+        //         }
+        //     }
+        await setupClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: {
+              times: 1
+            },
+            data: {
+              errorCode: 123,
+              failCommands: ['aggregate']
+            }
+          });
+      });
+
+      it('expects an error and a command failed event', async function () {
+        // Use ``encryptedClient`` to run an aggregate on ``db.decryption_events``.
+        // Expect an exception to be thrown from the command error. Expect a CommandFailedEvent.
+        const collection = encryptedClient.db('').collection('decryption_events');
+        let result;
+        try {
+          result = await collection.aggregate([]).toArray();
+        } catch (error) {
+          expect(error.code).to.equal(123);
+        }
+        expect(result).to.not.exist;
```

Comment:
this is a nitpick so feel free to ignore it - rather than asserting that result is undefined, we could use an `expect.fail` between lines 1561 and 1562.  That will fail the test if an error isn't thrown by `collection.aggregate`.

Reply:
Updated to use `expect.fail`